tests/helpers.py

```diff
@@ -29,10 +29,10 @@ def create_unique_resource_name(resource_description, resource_type):
     Returns:
         str: A unique name
     """
-    current_date_time = (
-        datetime.datetime.now().strftime("%d%H%M%S%f")
-    )
-    return f"{resource_type}-{resource_description[:23]}-{current_date_time[:10]}"
+    randnum1 = randrange(100, 1000000)
+    randnum2 = randrange(1000, 10000)
+    randnum = randnum1 * randnum2
+    return f"{resource_type}-{resource_description[:23]}-{randnum}"
```

Comment:
I am proposing to keep the same length of the random number and keep the date timestamp as it's sometime useful for easy identification when the object got created and you can directly watch the log with this timestamp.

```python
randnum = (
    f'{datetime.datetime.now().strftime("%d%H%M%S")}'
    f'{randrange(1, 10**5):05}'
)
```
Example of 10 numbers generated in for loop:
1309275981707
1309275933379
1309275951637
1309275946251
1309275968797
1309275955706
1309275989516
1309275972782
1309275994294
1309275964369

Reply:
I would add to this that we should separate the timestamp from the random numbers with `_` or `-`. This way the timestamp can easily be identified.

Reply:
when running using threadpool the first 8 digits are same which is not what we want, in less than a second we can invoke 1000's of pvc creation rather a truly random number is needed.

Reply:
if we need a truly unique string why don't we use `uuid`? If using the timestamp isn't a requirement I think this would be a better option for us.

```
In [30]: import uuid

In [31]: uuid.uuid4().hex
Out[31]: 'a0e3c31cbb6641aaa30432f59340d59f'
```

Reply:
Agree with @clacroix12 , If we are not really concerned about timestamp, UUID would be better rather than random number.

Reply:
uuid depends on random program, but I really don't have a preference where it comes from(as long as its thread safe), from the python's own documentation 

```
Python uses the Mersenne Twister as the core generator. It produces 53-bit precision floats and has a period of 2**19937-1. The underlying implementation in C is both fast and threadsafe
```

Reply:
You can check to see if your uuid was generated in a thread safe way with the `.is_safe` attribute.

```
my_uuid = uuid.uuid4()
my_uuid.is_safe
```

I did some local testing with threads and never hit any conflicts, however the `is_safe` attribute was always unknown for me. Not sure if that is a system issue or what, but you might be able to get different results?

Reply:
I will use uuid4() and rerun the test with 500+ threads, if I dont hit issues I will just change it.

Reply:
I see this is working fine, there are parts in our framework which is causing some issues while testing > 100 but nothing related to uuid part.